libs/ledgerjs/packages/hw-app-btc/src/BtcNew.ts

```diff
@@ -453,3 +443,57 @@ function accountTypeFromArg(
   if (arg.segwit) return new p2wpkhWrapped(psbt, masterFp);
   return new p2pkh(psbt, masterFp);
 }
+
+/*
+  The new protocol only allows standard path.
+  Standard paths are (currently):
+  M/44'/(1|0)'/X'
+  M/49'/(1|0)'/X'
+  M/84'/(1|0)'/X'
+  M/86'/(1|0)'/X'
+  M/48'/(1|0)'/X'/Y'
+  followed by "", "(0|1)", or "(0|1)/b", where a and b are 
+  non-hardened. For example, the following paths are standard
+  M/48'/1'/99'/7'
+  M/86'/1'/99'/0
+  M/48'/0'/99'/7'/1/17
+  The following paths are non-standard
+  M/48'/0'/99'           // Not deepest hardened path
+  M/48'/0'/99'/7'/1/17/2 // Too many non-hardened derivation steps
+  M/199'/0'/1'/0/88      // Not a known purpose 199
+  M/86'/1'/99'/2         // Change path item must be 0 or 1
+*/
+function isPathNormal(path: string): boolean {
+  //path is not deepest hardened node of a standard path or deeper, use BtcOld
+  const h = 0x80000000;
+  const pathElems = pathStringToArray(path);
```

Comment:
```suggestion
  const [root, hardPath, softPath, foo, bar] = pathStringToArray(path);
```

if possible could you deconstruct and name the variable that are returned instead of using `path[0], path[1], ...` so that it's clearer for everyone what each part correspond to

Reply:
This piece of code comes from legacy code. I think your proposal is not possible because from here we can't deduce the length of elements return from the method "pathStringToArray". The "path" here can be an arbitrary string and it can return an array with arbitrary length